cep-purls.md

```diff
@@ -0,0 +1,192 @@
+<table>
+<tr><td> Title </td><td> Add package-urls to PackageRecord </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Bas Zalmstra &lt;bas@prefix.dev&gt;, Pavel Zwerschke &lt;pavelzw@gmail.com&gt; </td></tr>
+<tr><td> Created </td><td> Nov 23, 2023</td></tr>
+<tr><td> Updated </td><td> Nov 23, 2023</td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda/ceps/pull/63 </td></tr>
+<tr><td> Implementation </td><td> https://github.com/prefix-dev/rattler-build/pull/1664, https://github.com/conda/rattler/pull/1303, https://github.com/conda/rattler/pull/1312 </td></tr>
+</table>
+
+## Abstract
+
+This CEP describes a change to the `PackageRecord` format and the corresponding `repodata.json` file to include `purls` (Package Urls) of repackaged packages to identify packages across multiple ecosystems.
+
+## Specification
+
+We propose to add the optional `purls: [string]` field to `PackageRecord`.
+To identify the repackaged package we use [PURL](https://github.com/package-url/purl-spec/) (Package URL), which implements a scheme for identifying packages that is meant to be portable across packaging ecosystems.
+
+```json
+{
+  ...
+  "pinject-0.14.1-pyhd8ed1ab_0.tar.bz2": {
+    "name": "pinject",
+    "version": "0.14.1",
+    "purls": ["pkg:pypi/pinject@0.14.1"],
+    ...
+  }
+  ...
+}
+```
+
+PURL is already supported by dependency-related tooling like SPDX (see [External Repository Identifiers in the SPDX 2.3 spec](https://spdx.github.io/spdx-spec/v2.3/external-repository-identifiers/#f35-purl)), the [Open Source Vulnerability format](https://ossf.github.io/osv-schema/#affectedpackage-field), and the [Sonatype OSS Index](https://ossindex.sonatype.org/doc/coordinates); not having to wait years before support in such tooling arrives is valuable.
+[PEP 725 (WIP)](https://peps.python.org/pep-0725) also proposes how to specify non-PyPi dependencies using PURLs.
+
+### PURLs in recipes
+
+PURLs can be specified in the recipe under `.about.purls` (implementation for [v1-recipes](cep-00014.md): [prefix-dev/rattler-build #1664](https://github.com/prefix-dev/rattler-build/pull/1664)):
+
+```yaml
+about:
+  # ...
+  # PURLs that this package belongs to
+  purls: [string (PURL enforced)]
+```
+
+Tools that build conda packages should add these packages to `index.json`:
+
+```json
+{
+  ...,
+  "purls": ["pkg:pypi/pinject@0.14.1"]
+}
+```
+
+### PURLs in repodata
+
+In `repodata.json`, PURLs should be an optional field.
+Tools that generate `repodata.json` (like `rattler-index` or `conda-index`) can decide whether to include it or not in `repodata.json`.
+
+Instead of writing the PURLs in `repodata.json`, these tools can decide to add PURLs to a separate `purls.json` like in [CEP 12](cep-0012.md):
```

Comment:
```suggestion
Instead of writing the PURLs in `repodata.json`, these tools MAY add PURLs to a separate `purls.json` file, like in [CEP 12](cep-0012.md), with the following schema:
```